parsing/lexer.mll

```diff
@@ -418,35 +467,40 @@ rule token = parse
   | ".~"
       { error lexbuf
           (Reserved_sequence (".~", Some "is reserved for use in MetaOCaml")) }
-  | "~" raw_ident_escape (lowercase identchar * as name) ':'
-      { LABEL name }
-  | "~" (lowercase identchar * as name) ':'
+  | "~" (identstart identchar * as name) ':'
       { check_label_name lexbuf name;
         LABEL name }
-  | "~" (lowercase_latin1 identchar_latin1 * as name) ':'
-      { warn_latin1 lexbuf;
+  | "~" (raw_ident_escape? as escape) (ident_ext as raw_name) ':'
```

Comment:
Nice simplification ! Though, does the lexing performances of labels matter ? I'd expect it to matter less than LIDENT and we could get rid of more rules.

Reply:
It matters less, but I wouldn't be surprised if some coding style ends up with a significant minority of label identifiers.